File: rust-surrealDB-Elm/back/src/services/queries.rs

pub const TABLE : &str = "table";
pub const LABEL : &str = "label";
pub const ID : &str = "id";
pub const EXIST_BY_LABEL: &str = r#"SELECT * FROM (
    SELECT * FROM count(
        (
            SELECT * FROM type::table($table)
            WHERE string::trim(string::lowercase(label)) = $label
        )
    )
)[0] > 0"#;

pub const EXIST_BY_ID: &str = r#"SELECT * FROM (
    SELECT * FROM count(
        (
            SELECT * FROM type::table($table)
            WHERE id = type::thing($table, $id)
        )
    )
)[0] > 0"#;

File: Bun-ElysiaJs-HTMX/src/back/storage/queries.ts

export const GET_ALL = 'SELECT * FROM todo'

export const INSERT = `INSERT INTO todo (id, label, completed)VALUES ($id, $label, $completed) returning *`

export const GET_BY_ID = 'SELECT * FROM todo WHERE id = $id'

export const UPDATE = `UPDATE todo SET completed = $completed, label = $label WHERE id = $id returning *`

export const DELETE = 'DELETE FROM todo WHERE id = $id'

export const EXIST_BY_ID = 'SELECT EXISTS(SELECT * FROM todo WHERE id = $id) as exist'

export const EXIST_BY_LABEL = 'SELECT EXISTS(SELECT * FROM todo WHERE label = $label) as exist'